graph/pkg/service/v0/driveitems.go

```diff
@@ -185,13 +240,13 @@ func (g Graph) getSpecialDriveItem(ctx context.Context, ID *storageprovider.Reso
 		g.logger.Error().Err(err).Str("ID", ID.OpaqueId).Msg("Could not get readme Item")
 		return nil
 	}
-	itemPath, err := g.getPathForDriveItem(ctx, ID)
+	itemPath, err := g.getPathForResource(ctx, ID)
 	if err != nil {
 		g.logger.Error().Err(err).Str("ID", ID.OpaqueId).Msg("Could not get readme path")
 		return nil
 	}
 	spaceItem.SpecialFolder = &libregraph.SpecialFolder{Name: libregraph.PtrString(itemName)}
-	spaceItem.WebDavUrl = libregraph.PtrString(baseURL.String() + path.Join(space.Root.OpaqueId, *itemPath))
+	spaceItem.WebDavUrl = libregraph.PtrString(baseURL.String() + path.Join(space.Root.StorageId, *itemPath))
```

Comment:
shouldn't this be the spaceid? as in `{spaceid}!{nodeid}` or `{storageid}!{opaqueid}` ... so this also works for shares where `{storageid} != {opaqueid}`.

Reply:
changed